src/number_theory/divisors.lean

```diff
@@ -226,8 +226,46 @@ begin
     simp [h] }
 end
 
+@[simp] lemma image_fst_divisors_antidiagonal :
+  (divisors_antidiagonal n).image prod.fst = divisors n :=
+by { ext, simp [has_dvd.dvd, eq_comm] }
```

Comment:
The combination of `simp` without `only` and `eq_comm` seems quite sensitive to `simp` order changes. If I understand correctly why you use `eq_comm`, maybe [dvd_iff_exists_eq_mul_left](https://leanprover-community.github.io/mathlib_docs/algebra/divisibility.html#dvd_iff_exists_eq_mul_left) is a better alternative?

Reply:
`dvd_iff_exists_eq_mul_left` commutes the multiplication, my problem is that I need to commute the equality. I think `simp` is pretty good at applying commutativity lemmas in a deterministic way that isn't influenced too much by lemma order.

Reply:
Would you be happier with this?
```suggestion
by { ext, simp [has_dvd.dvd, @eq_comm ℕ n (_ * _)] }
```

Reply:
That would nicely reassure me, thanks! :)